odoo/addons/base/models/res_country.py

```diff
@@ -40,6 +44,18 @@ class Country(models.Model):
         string='Country Code', size=2,
         required=True,
         help='The ISO country code in two chars. \nYou can use this field for quick search.')
+    iso_alpha3 = fields.Char(
+        string='Alpha-3 Code', size=3,
+        compute='_compute_iso_alpha3', store=True, readonly=False,
+        required=True, precompute=True,
+        help='The ISO country code in three chars.\nYou can use this field for quick search.'
+    )
+    iso_num = fields.Char(
+        string='Numeric Code', size=3,
+        compute='_compute_iso_num', store=True, readonly=False,
+        required=True, precompute=True, 
+        help='The ISO country code in three digits.\nYou can use this field for quick search.'
+    )
```

Comment:
Put the data in the XML file and call it a day.  I don't get the point of putting pure data in the code.

Reply:
It is a mapping that shouldn't be modified by the user
* if you deleted a country, it is no easy to recreate it (without updating the whole module...)
* the data is easy to update because it follows the ISO datasheet, and only that. No unrelated noise
* it takes less loc because one line per country
* it was easier for me to write than to edit the XML file.

But sure, if you think that it is better to do it the tedious way and less user friendly way, I'll do it.

Reply:
It's only data, please don't overengineer it.
The only code that's necessary are the unique constraints and the override of `_name_search()`.

Reply:
I actually have to remove the unique constraints.
For instance, both Great Britain and Northern Ireland have `GBR` as alpha-3 code, but the alpha-2 is respectively `GB` and `XI` since Brexit.